src/base/rss/rss_article.cpp

```diff
@@ -30,38 +30,52 @@
 
 #include "rss_article.h"
 
+#include <stdexcept>
 #include <QJsonObject>
 #include <QVariant>
 
 #include "rss_feed.h"
 
-const QString Str_Id(QStringLiteral("id"));
-const QString Str_Date(QStringLiteral("date"));
-const QString Str_Title(QStringLiteral("title"));
-const QString Str_Author(QStringLiteral("author"));
-const QString Str_Description(QStringLiteral("description"));
-const QString Str_TorrentURL(QStringLiteral("torrentURL"));
-const QString Str_Torrent_Url(QStringLiteral("torrent_url"));
-const QString Str_Link(QStringLiteral("link"));
-const QString Str_News_Link(QStringLiteral("news_link"));
-const QString Str_IsRead(QStringLiteral("isRead"));
-const QString Str_Read(QStringLiteral("read"));
-
 using namespace RSS;
 
-Article::Article(Feed *feed, QString guid, QDateTime date, QString title, QString author
-                 , QString description, QString torrentUrl, QString link, bool isRead)
+const QString Article::KeyId(QStringLiteral("id"));
+const QString Article::KeyDate(QStringLiteral("date"));
+const QString Article::KeyTitle(QStringLiteral("title"));
+const QString Article::KeyAuthor(QStringLiteral("author"));
+const QString Article::KeyDescription(QStringLiteral("description"));
+const QString Article::KeyTorrentURL(QStringLiteral("torrentURL"));
+const QString Article::KeyLink(QStringLiteral("link"));
+const QString Article::KeyIsRead(QStringLiteral("isRead"));
+
+Article::Article(Feed *feed, const QVariantHash &varHash)
     : QObject(feed)
     , m_feed(feed)
-    , m_guid(guid)
-    , m_date(date)
-    , m_title(title)
-    , m_author(author)
-    , m_description(description)
-    , m_torrentURL(torrentUrl)
-    , m_link(link)
-    , m_isRead(isRead)
+    , m_guid(varHash.value(KeyId).toString())
+    , m_date(varHash.value(KeyDate).toDateTime())
+    , m_title(varHash.value(KeyTitle).toString())
+    , m_author(varHash.value(KeyAuthor).toString())
+    , m_description(varHash.value(KeyDescription).toString())
+    , m_torrentURL(varHash.value(KeyTorrentURL).toString())
+    , m_link(varHash.value(KeyLink).toString())
+    , m_isRead(varHash.value(KeyIsRead, false).toBool())
+    , m_data(varHash)
+{
+    // If item does not have a guid, fall back to some other identifier
+    if (m_guid.isEmpty())
+        m_guid = varHash.value(KeyTorrentURL).toString();
+    if (m_guid.isEmpty())
+        m_guid = varHash.value(KeyTitle).toString();
+    if (m_guid.isEmpty()) throw std::logic_error("Bad RSS Article data");
```

Comment:
Yes, you have to throw `std::runtime_error` or its descendant here (see my other comment to `catch`).